### ISSUE TITLE
glob not working

### ISSUE THREAD

**Issue by pixelass** [stylelint/stylelint#1970]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

bug

> Which rule, if any, is this issue related to?

global

> What CSS is needed to reproduce this issue?

none

> What stylelint configuration is needed to reproduce this issue?

none

> Which version of stylelint are you using?

`7.4.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint  src/**/*.css`

> What did you expect to happen?

all files in all directories are linted

> What actually happened (e.g. what warnings or errors you are getting)?

only one level (`src/*/*.css`) is linted

```shell
# glob: src/**/*.css
src/style.css -> NOT linted
src/foo/style.css -> linted
src/bar/style.css -> linted
src/foo/bar/style.css -> NOT linted
src/foo/bar/style.css -> NOT linted
```

**Comment by davidtheclark** [comment]
@pixelass: Try putting quotation marks around your glob. What happens then?

**Comment by pixelass** [comment]
tried it initially. it causes an error

```shell
> stylelint 'packages/**/*.css'

SyntaxError: Unexpected token ? in JSON at position 1328
```

**Comment by davidtheclark** [comment]
@pixelass That's pretty weird ... not something we've seen before or other users are running into. Is the syntax error correct — do you have a misplaced `?`? Could you share a gist or repo with your files?

**Comment by evilebottnawi** [comment]
@davidtheclark @jeddy3 what about add label `triage` (as `eslint`), because some issues are not discussion, but we have to show that see this?

**Comment by davidtheclark** [comment]
@evilebottnawi I think if we don't yet have a good label for it, e.g. not sure if it's a bug or not like this one, we can just leave it unlabeled, right?

**Comment by ntwb** [comment]
Historically we've used `discussion` pretty much for this purpose, then changed the labels once we have determined if it's a bug

**Comment by pixelass** [comment]
thank you everybody.
@davidtheclark There was an issue with a css file. I am now excluding it and the pattern works in `'` quotation marks.
I will look into it again to see what was causing the error. 
closing this for now since my issue was resolved by following your suggestions.

**Comment by davidgilbertson** [comment]
Same issue here

node: 6.6.0
npm: 3.10.3
stylelint: 7.6.0

Can confirm that when calling this from an npm script.

- `myApp/css/*.scss` will look only in the `css` (correct)
- `myApp/css/**.*.scss` will look only in directories immediately under `css` (but not files in the top level of the `css` directory) (incorrect)

It's actually a sneaky issue because _most_ of my scss files are only one level down, so it appeared to be working fine. But all my scss files two levels or more down were being skipped quietly.

Changing my script from `"lint-sass": "stylelint myApp/css/**/*.scss",` to `"lint-sass": "stylelint 'myApp/**/*.scss'",` (adding single quotes _and_ pointing one level further up the tree) worked.

I'd say this is either a bug or would be great to see as a caveat in the CLI docs.

**Comment by pixelass** [comment]
@davidgilbertson I belive this is wrong: `myApp/css/**.*.scss`

Try `myApp/css/**/*.scss`

**Comment by evilebottnawi** [comment]
@davidgilbertson can you create example repo with reproduced error?

**Comment by davidgilbertson** [comment]
@evilebottnawi does that mean that where you're testing, if you do an npm script such as `"lint-sass": "stylelint 'myApp/css/**/*.scss'", it runs on files in the `css` directory and two or three levels deep?

**Comment by evilebottnawi** [comment]
@davidgilbertson yep, and deep and deep and deep and forever deep

**Comment by davidgilbertson** [comment]
Sorry I was getting mixed up, to clarify:

- glob with **no single quotes** around it: only includes one level down, doesn't go deep
- glob **with single quotes** around it: goes deep, but doesn't select the top level folder (e.g. `"lint-sass": "stylelint 'myApp/css/**/*.scss'"` will not lint the file `myApp/css/main.scss`.

Do you get the behaviour of the second instance? Or does a top level file get linted with that glob pattern? If not I'll try and do a concise repo on Monday.

**Comment by evilebottnawi** [comment]
@davidgilbertson provide please, os, node, npm verison

**Comment by chrisvfritz** [comment]
On at least one Windows machine, it wanted double quotes around the glob instead of single quotes. May help others hopefully.
